src/wechaty/user/contact.py

```diff
@@ -132,25 +131,49 @@ async def find_all(cls: Type[Contact],
         log.info('find_all() <%s, %s>', cls, query)
 
         contact_ids = await cls.get_puppet().contact_list()
+        # filter Contact by contact id to make sure its valid if contact_id.startswith('wxid_')
         contacts = [Contact.load(contact_id) for contact_id in contact_ids]
 
         # load contact parallel using asyncio.gather method
-        await asyncio.gather(*[contact.ready() for contact in contacts])
 
         # async load
         batch_size = 16
-        # slice contacts by batch_size
-        contacts = contacts[:batch_size * (len(contacts) // batch_size)]
-
+        batch_index = 0
         contact_result_list: List[Contact] = []
-
-        for contact in contacts:
+        # slice contacts by batch_size
+        while batch_index * batch_size < len(contacts):
+            batch_contactList = contacts[batch_index * batch_size: (batch_index + 1) * batch_size]
             try:
-                await contact.ready()
-                contact_result_list.append(contact)
+                await asyncio.gather(*[contact.ready() for contact in batch_contactList])
+                contact_result_list += batch_contactList
             except RuntimeError as exception:
                 log.info('load contact occur exception: %s', exception.args)
+            batch_index = batch_index + 1
```

Comment:
There is an improvement for the code. Contact list data has been saved twice, `contacts` and `contact_result_list`.  `id(contacts[0])` is equal to `id(contact_result_list[0])`, you can test it. 

How do you think about it ?

Reply:
how about use one line `await asyncio.gather(*[contact.ready() for contact in contacts])` instead , and avoid repeated request for contact information

Reply:
I think it's fine to use one line. 

First, using `batch_size`, `batch_index`  is to make `IO` more relaxed. And `asyncio.gather` is to schedule the `IO` in one process. What's more, I think `asyncio.gather` can be better than `batch_size`.  ✨